homeassistant/components/lamarzocco/calendar.py

```diff
@@ -85,28 +111,46 @@ def _async_get_calendar_event(self, date: datetime) -> CalendarEvent | None:
         """Return calendar event for a given weekday."""
 
         # check first if auto/on off is turned on in general
-        # because could still be on for that day but disabled
-        if self.coordinator.lm.current_status["global_auto"] != "Enabled":
+        if not self.wake_up_sleep_entry.enabled:
             return None
 
         # parse the schedule for the day
-        schedule_day = self.coordinator.lm.schedule[date.weekday()]
-        if schedule_day["enable"] == "Disabled":
+
+        if DAY_OF_WEEK[date.weekday()] not in self.wake_up_sleep_entry.days:
             return None
-        hour_on, minute_on = schedule_day["on"].split(":")
-        hour_off, minute_off = schedule_day["off"].split(":")
+
+        hour_on, minute_on = self.wake_up_sleep_entry.time_on.split(":")
+        hour_off, minute_off = self.wake_up_sleep_entry.time_off.split(":")
+
+        # convert 24:00 to 23:59:59.999999
+        second_on = 0
+        second_off = 0
+        microsecond_on = 0
+        microsocond_off = 0
+
+        if hour_on == "24":
+            hour_on = "23"
+            minute_on = "59"
+            second_off = 59
+            microsecond_on = 999999
+        if hour_off == "24":
+            hour_off = "23"
+            minute_off = "59"
+            second_off = 59
+            microsocond_off = 999999
+
         return CalendarEvent(
             start=date.replace(
                 hour=int(hour_on),
                 minute=int(minute_on),
-                second=0,
-                microsecond=0,
+                second=second_on,
+                microsecond=microsecond_on,
             ),
             end=date.replace(
                 hour=int(hour_off),
                 minute=int(minute_off),
-                second=0,
-                microsecond=0,
+                second=second_off,
+                microsecond=microsocond_off,
             ),
```

Comment:
What's this for

Reply:
Different date representations. LM knows 24:00, but for HA it's 0:00 on the next day, so I'm just moving it a microsecond earlier for HA

Reply:
made it clearer/better